src/test/java/org/opensearch/performanceanalyzer/store/rca/hotcpu/ThreadMetricsSlidingWindowTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.store.rca.hotcpu;
+
+
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.opensearch.performanceanalyzer.rca.GradleTaskForRca;
+import org.opensearch.performanceanalyzer.rca.store.rca.hot_node.ThreadMetricsRca.ThreadMetric;
+import org.opensearch.performanceanalyzer.rca.store.rca.hot_node.ThreadMetricsRca.ThreadMetricsSlidingWindow;
+
+@Category(GradleTaskForRca.class)
+public class ThreadMetricsSlidingWindowTest {
```

Comment:
We should also add unit tests for ThreadMetricsRca for different scenarios. And adding unit tests around sliding window logic to catch any issues.